ducktape/cluster/remoteaccount.py

```diff
@@ -62,6 +63,8 @@ def _can_ping_url(self, url, headers):
     def ssh_command(self, cmd):
         if self.local:
             return cmd
+        if self.sudo_user is not None:
```

Comment:
@lindong28 
- I'm leaning again toward not specifying in RemoteAccount unless there is a clear use-case 
- I think in general we don't want to ssh in as root, and only run as root where necessary; in the few cases where we need increased privileges, we pretty much just directly run: `node.account.ssh("sudo cmd")`

In general, I think my and Ewen's biggest concerns around this patch are that we don't understand exactly your use-case and why it's the most appropriate solution. Would you mind sketching that out?